spacy/tests/pipeline/test_span_finder.py

```diff
@@ -0,0 +1,272 @@
+
+    assert doc.spans[SPANS_KEY]
+    assert len(doc.spans[SPANS_KEY]) == span_count
+
+    # Assert below will fail when max_length is set to 0
+    if max_length is None:
+        max_length = float("inf")
+    if min_length is None:
+        min_length = 1
+
+    assert all(min_length <= len(span) <= max_length for span in doc.spans[SPANS_KEY])
+
+
+def test_span_finder_suggester():
+    nlp = Language()
+    docs = [nlp("This is an example."), nlp("This is the second example.")]
+    docs[0].spans[SPANS_KEY] = [docs[0][3:4]]
+    docs[1].spans[SPANS_KEY] = [docs[1][3:5]]
+    span_finder = nlp.add_pipe("span_finder", config={"spans_key": SPANS_KEY})
+    nlp.initialize()
+    # Setting bias terms to 1000 to always make it predict something.
+    scorer_model = span_finder.model.get_ref("scorer")
+    bias = scorer_model.layers[0].get_param("b")
+    bias += 1000
+    scorer_model.layers[0].set_param("b", bias)
+    span_finder.set_annotations(docs, span_finder.predict(docs))
+
+    suggester = registry.misc.get("spacy.span_finder_suggester.v1")(
+        spans_key=SPANS_KEY
+    )
+
+    candidates = suggester(docs)
+
+    span_length = 0
+    for doc in docs:
+        span_length += len(doc.spans[SPANS_KEY])
+
+    assert span_length == len(candidates.dataXd)
+    assert type(candidates) == Ragged
+    assert len(candidates.dataXd[0]) == 2
+
+
+def test_overfitting_IO():
+    # Simple test to try and quickly overfit the span_finder component - ensuring the ML models work correctly
+    fix_random_seed(0)
+    nlp = English()
+    span_finder = nlp.add_pipe("span_finder", config={"spans_key": SPANS_KEY})
+    train_examples = make_examples(nlp)
+    optimizer = nlp.initialize(get_examples=lambda: train_examples)
+    assert span_finder.model.get_dim("nO") == 2
+
+    for i in range(50):
+        losses = {}
+        nlp.update(train_examples, sgd=optimizer, losses=losses)
+    assert losses["span_finder"] < 0.001
+
+    # test the trained model
+    test_text = "I like London and Berlin"
+    doc = nlp(test_text)
+    spans = doc.spans[span_finder.spans_key]
+    assert len(spans) == 3
+    assert set([span.text for span in spans]) == {
+        "London",
+        "Berlin",
+        "London and Berlin",
+    }
+
+    # Also test the results are still the same after IO
+    with make_tempdir() as tmp_dir:
+        nlp.to_disk(tmp_dir)
+        nlp2 = util.load_model_from_path(tmp_dir)
+        doc2 = nlp2(test_text)
+        spans2 = doc2.spans[span_finder.spans_key]
+        assert len(spans2) == 3
+        assert set([span.text for span in spans2]) == {
+            "London",
+            "Berlin",
+            "London and Berlin",
+        }
+
+    # Test scoring
+    scores = nlp.evaluate(train_examples)
+    assert f"span_finder_{span_finder.spans_key}_f" in scores
+    # XXX Its not perfect 1.0 F1 because we want it to overgenerate for now.
+    assert scores[f"span_finder_{span_finder.spans_key}_f"] == 0.4
```

Comment:
Maybe to make it a little more clear what we're going for with the overfitting?

```suggestion
    assert scores[f"span_finder_{span_finder.spans_key}_f"] == 0.4
    assert scores[f"span_finder_{span_finder.spans_key}_r"] == 1.0
```